HIPs/incubation/waterfall.md

```diff
@@ -0,0 +1,236 @@
+- Sergii Grybniak, PhD [<sergii@bluewave.globall>](mailto:sergii@bluewave.global)
+- Oleksandr Nashyvan [<on@waterfall.network>](mailto:on@waterfall.network)
+- Yevhen Leonchyk, PhD [<leochyk@onu.edu.ua>](mailto:leochyk@onu.edu.ua)
+- Igor Mazurok, PhD [<mazurok@onu.edu.ua>](mailto:mazurok@onu.edu.ua)
+- Ruslan Shanin, PhD [<ruslanshanin@onu.edu.ua>](mailto:ruslanshanin@onu.edu.ua)
+- Vladimir Mezin, PhD [<vladimir@waterfall.network>](mailto:vladimir@waterfall.network)
+
+# Abstract
+
+[Waterfall](https://waterfall.network/) is a highly scalable smart contract network for the development of various
+decentralized applications (DApps).
+The distributed protocol is based on DAG (Directed Acyclic Graph) technology, with a 2-level fast finality
+Proof-of-Stake (PoS) consensus protocol, in which millions of validators can participate.
+It challenges the trilemma of scalability, security, and decentralization by bringing the combination of scalability and
+decentralization on a new level.
+
+Waterfall integrates with the Ethereum Virtual Machine (EVM) that all Ethereum-based apps can be run on this network.
+Therefore, Waterfall can ensure a favorable environment for the provision and consumption of a wide spectrum of
+enterprise-class services for business and social activities, in a convenient format within the framework of
+a decentralized network.
+
+# Context
+
+Despite its potential, DLT is still an innovative technology, and some challenges need to be addressed.
+One of the main obstacles is scalability, as the current generation of DLTs can only handle a limited number of
+transactions per second without compromising on security and/or increased overall network centralization – 
+the so-called trilemma of scalability, security, and decentralization.
+
+Vitalik Buterin, the creator of Ethereum, the currently most decentralized smart contract network,
+formulated the blockchain trilemma: of the three chief characteristics of DLT – decentralization, security, and
+scalability – a distributed system can only have two.
+In many cases, this trilemma boils down to a simple tradeoff between scalability and decentralization,
+since security in fintech is non-negotiable.
+Vitalik Buterin clearly highlighted all the advantages and disadvantages in his article "[Endgame](https://vitalik.eth.limo/general/2021/12/06/endgame.html),"
+which can be summarized as "distributed systems cannot be scalable and decentralized at the same time, and that is fine."
+
+![Blockchain Trilemma](../images/waterfall/trilemma.png "Blockchain Trilemma")
+
+As DLT continues to gain popularity and widespread adoption, it is essential to explore new solutions and refine
+existing ones to address the scalability challenge.
+Nowadays, increasing scalability, security, and decentralization remain a primary concern for the mass adoption of
+decentralized services, especially at the enterprise level.
+
+# Dependent Projects
+
+None.
+Waterfall was derived from Prysm and Go-Ethereum.
```

Comment:
We were using the original licenses for the source code we have taken and Apache 2.0 for the additional files we created. For the files that were edited, we preserved both original and new licenses. Please let us know if we should do it differently, and it will be fixed.